### DISCUSSION TITLE
Request: a build option to disable the web UI embedded into the llama-server executable

### DISCUSSION THREAD

**Discussion by SamuelTallet** [ggml-org/llama.cpp#10986] [Ideas]
Hello everyone,

What do you think about adding  a CMake option like `LLAMA_SERVER_NO_WEBUI` (default: `OFF`) [here](https://github.com/ggerganov/llama.cpp/blob/9ba399dfa7f115effc63d48e6860a94c9faa31b2/examples/server/CMakeLists.txt#L4)?

This way, the `llama-server` binary could stay light for those who don't need the embedded web UI.

In other words, with this option `ON`, _index.html.gz_ file would not be embedded into `llama-server`.

**Comment by EloxFire** [comment]
Great idea ! Would be very useful in some cases and keep bundle size lower.

**Comment by woof-dog** [comment]
Also would be nice to have a CMake option for a path to a downloaded version of the python interpreter locally so it's not downloaded over the web at runtime and can be used offline...

@SamuelTallet from what I've learned the llama-server maintainer doesn't have much time for feature requests like this so you need to make the PR yourself or it won't be done. Fair enough for open source I guess.

**Comment by kushagharahi** [comment]
Put up https://github.com/ggml-org/llama.cpp/pull/20158 for this request

**Reply by SamuelTallet** [reply]
Thanks a lot! :)
